### PR TITLE
AZP/RELEASE: Fixes for v1.14 release

### PR DESCRIPTION
**Pull request by Al3xR01**
## What
Manually cherry-pick fixes to resolve various CI failures.

### CODE REVIEW

**Review by artemry-nv — APPROVED** [review]

**Comment by artemry-nv** [comment]
Have you verified test release on your fork with these changes?

**Comment by Al3xR01** [comment]
> Have you verified test release on your fork with these changes?

For the most part:
https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=57018&view=results
The  Maven publish stage uses some pipeline-specific secrets that I can't replicate.

**Review by brminich — DISMISSED** [review]

**Comment by brminich** [comment]
@yosefe, @artemry-nv

**Comment by yosefe** [comment]
@Al3xR01  the build failure seems relevant,
also can you pls change the commit title to "AZP/RELEASE: ..."? see
https://github.com/openucx/ucx/blob/88d4bb96e189fd82e77ea2cfb246192ab4944365/buildlib/azure-pipelines-release.yml#L42
it will trigger release pipeline for this PR so we can see if it works

**Comment by Al3xR01** [comment]
@yosefe Looks good:
https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=57027&view=results

File: debian/rules.in [hunk 1]
```diff
@@ -16,7 +16,7 @@ CUDA_OPT = $(shell echo ${DEB_BUILD_PROFILES} | grep -qw cuda \
 	dh $@ 
 
 override_dh_auto_configure:
-	./contrib/configure-release-mt --prefix=/usr \
+	../contrib/configure-release-mt --prefix=/usr \
```
Comment (brminich) [inline debian/rules.in:19]: some tests fail because of this change

Comment (Al3xR01) [reply debian/rules.in:19]: @brminich Debian builds [fail ](https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=57005&view=logs&j=698c344f-0db0-55eb-b02a-d42dfa778b4b&t=581722d4-d92c-5e13-274b-3ed82c756e80&l=41)without this change, so maybe we should fix these tests.

Comment (brminich) [reply debian/rules.in:19]: isn't [this](https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=57028&view=logs&j=a1437434-5193-5f40-7f41-b69248d83fa2&t=e254882d-22eb-5826-ce29-e3e5473c5857&l=37270) related?

Comment (Al3xR01) [reply debian/rules.in:19]: Resolved by the `sed` hack for now.

**Comment by brminich** [comment]
@yosefe, can you please review?

**Review by brminich — APPROVED** [review]

**Review by yosefe — APPROVED** [review]

**Pull request merged**